GTG/core/timer.py

```diff
@@ -70,15 +72,64 @@ def time_changed(self):
         self.timeout_source = GObject.timeout_add_seconds(secs_to_refresh,
                                                           self.emit_refresh)
 
-    def set_configuration(self, refresh_hour, refresh_min):
+    def set_configuration(self, time):
+        self.config.set('hour', time.hour)
+        self.config.set('min', time.minute)
+
+    def get_configuration(self):
         try:
-            datetime.time(int(refresh_hour), int(refresh_min))
-        except (ValueError, TypeError):
-            raise ValueError("{}:{} is not a valid time".format(refresh_hour,
-                                                                refresh_min))
+            return datetime.time(int(self.config.get('hour')),
+                                 int(self.config.get('min')))
+        except(ValueError):
+            Log.error("Invalid time values. Try using HH:MM or am/pm format")
+            return datetime.time(0, 0)
 
-        self.config.set('hour', refresh_hour)
-        self.config.set('min', refresh_min)
+    def parse_time(self, time):
+        """
+        This function parses user input when setting
+        start of the day in general preferences.
+        input:  @time               user's input as string
 
-    def get_configuration(self):
-        return self.config.get('hour'), self.config.get('min')
+        output: @hour               default format for hours
+                @minute             default format for minutes
+                @ValueError         raised if @time is not valid
+        """
+        time = time.lower().strip()
```

Comment:
Can you please do the .lower() only on the match? the fallback call of strptime might have an issue if lower() was applied